cloudsdk/src/main/java/io/particle/android/sdk/cloud/ParticleDevice.java

```diff
@@ -406,6 +406,31 @@ public void executeFlashingChange() throws RetrofitError {
         });
     }
 
+    @WorkerThread
+    public void flashCodeFile(final File file) throws ParticleCloudException {
+        performFlashingChange(new FlashingChange() {
+            @Override
+            public void executeFlashingChange() throws RetrofitError {
+                mainApi.flashFile(deviceState.deviceId,
+                        new TypedFile("multipart/form-data", file));
+            }
+        });
+    }
+
+
+    @WorkerThread
```

Comment:
You have two methods called `flashCodeFile()`.  One of them takes a `File`, the other takes an `InputStream`.

Item 40 from Effective Java (2nd edition) says:
> Don’t got overboard in providing convenience methods.  Every method should “pull its weight.” […] **When in doubt, leave it out.**

I'm not convinced that we need both the `File` and `InputStream` versions of this method.  While it's technically less flexible, it seems to me that in practice, the `File` method would get 100x the usage (or more) than the `InputStream` version.

Do you have any specific examples where you've needed an `InputStream` instead of a `File`?